File: elearnia-backend/src/main/java/com/elearnia/entities/Lesson.java
package com.elearnia.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Entity
@Table(name = "lessons")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "course"})
public class Lesson {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Lob
    private String description; // Utilise TEXT pour permettre du contenu long (texte, code, tableaux)

    private String videoUrl;  // URL de la vidéo (YouTube, Vimeo, ou URL directe)

    private Integer duration; // Durée en minutes

    private Integer orderIndex; // Ordre d'affichage dans le cours

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Course course;
}

File: front elearnia Flutter/elearnia_app/lib/models/lesson.dart
class Lesson {
  final int id;
  final String title;
  final String? description;
  final String videoUrl;
  final int? duration; // en minutes
  final int orderIndex;

  const Lesson({
    required this.id,
    required this.title,
    this.description,
    required this.videoUrl,
    this.duration,
    required this.orderIndex,
  });

  factory Lesson.fromJson(Map<String, dynamic> json) {
    return Lesson(
      id: json['id'] as int,
      title: json['title']?.toString() ?? '',
      description: json['description']?.toString(),
      videoUrl: json['videoUrl']?.toString() ?? '',
      duration: json['duration'] as int?,
      orderIndex: json['orderIndex'] as int? ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'videoUrl': videoUrl,
      'duration': duration,
      'orderIndex': orderIndex,
    };
  }
}


